test/unit/relayServerUtils.test.ts

```diff
@@ -0,0 +1,446 @@
+        const dataWhenTransferingZero = dataWhenTransfer.replace(
+          tokenAmountToTransferAsHex,
+          '0'.repeat(64)
+        );
+
+        const request = createRequest({
+          data: dataWhenTransferingZero,
+        });
+
+        const config = {
+          disableSponsoredTx: true,
+          sponsoredDestinations: [],
+          gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+          transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
+        } as unknown as AppConfig;
+
+        const fee = await relayServerUtils.calculateFee(
+          request,
+          BigNumber.from(FAKE_ESTIMATION_BEFORE_FEES),
+          config
+        );
+
+        expect(fee.toString()).to.eq('0');
+      });
+
+      it('Should not charge extra fees when a transferFrom() with value = 0 is being relayed', async function () {
+        const dataWhenTransferingZero = dataWhenTransferFrom.replace(
+          tokenAmountToTransferAsHex,
+          '0'.repeat(64)
+        );
+
+        const request = createRequest({
+          data: dataWhenTransferingZero,
+        });
+
+        const config = {
+          disableSponsoredTx: true,
+          sponsoredDestinations: [],
+          gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+          transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
+        } as unknown as AppConfig;
+
+        const fee = await relayServerUtils.calculateFee(
+          request,
+          BigNumber.from(FAKE_ESTIMATION_BEFORE_FEES),
+          config
+        );
+
+        expect(fee.toString()).to.eq('0');
+      });
+
+      it('Should not charge extra fees when transferFee = 0 and a transfer is being relayed', async function () {
+        const request = createRequest({
+          data: dataWhenTransfer,
+        });
+
+        const config = {
+          disableSponsoredTx: true,
+          sponsoredDestinations: [],
+          gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+          transferFeePercentage: 0,
+        } as unknown as AppConfig;
+
+        const fee = await relayServerUtils.calculateFee(
+          request,
+          BigNumber.from(FAKE_ESTIMATION_BEFORE_FEES),
+          config
+        );
+
+        expect(fee.toString()).to.eq('0');
+      });
+
+      it('Should not charge extra fees when transferFee = 0 and a transferFrom is being relayed', async function () {
+        const request = createRequest({
+          data: dataWhenTransferFrom,
+        });
+
+        const config = {
+          disableSponsoredTx: true,
+          sponsoredDestinations: [],
+          gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+          transferFeePercentage: 0,
+        } as unknown as AppConfig;
+
+        const fee = await relayServerUtils.calculateFee(
+          request,
+          BigNumber.from(FAKE_ESTIMATION_BEFORE_FEES),
+          config
+        );
+
+        expect(fee.toString()).to.eq('0');
+      });
+    });
+
+    describe('Gas fee scenarios when not sponsored', function () {
```

Comment:
`when not sponsored` description should be included in the outer description, so it's not necessary here.
```suggestion
    describe('With gas fee scenarios', function () {
```

Reply:
I have group them in a general describe